[5.3] Use \Doctrine\DBAL\Driver\PDOConnection if available

mzur opened the pull request:
Follow-up of #14359 to target the 5.3 branch.

GrahamCampbell commented on src/Illuminate/Database/Connectors/Connector.php:
```diff
@@ -3,6 +3,7 @@
 namespace Illuminate\Database\Connectors;
 
 use PDO;
+use Doctrine\DBAL\Driver\PDOConnection;
```
Please sort by length.

mzur committed:
Use \Doctrine\DBAL\Driver\PDOConnection if available

If doctrine/dbal is installed the PDOConnection class provided by
the package should be used instead of just PDO. Else e.g. detecting
the database version doesn't work.

josebailo commented:
Is this fix still working? I want to rename a column in a table with a json column but throws me the error:

> [Doctrine\DBAL\DBALException]
> Unknown database type json requested, Doctrine\DBAL\Platforms\MySQL57Platform may not support it.

Even it doesn't let me change any column...

mzur commented:
Looks like DBAL doesn't support the JSON type for MySQL yet (see [here](https://github.com/doctrine/dbal/pull/2266). It should work with PostgreSQL.
